### ISSUE TITLE
Dependabot doesn't update the PR when the master branch has a force-push

### ISSUE THREAD

**Issue by jyn514** [dependabot/dependabot-core#4722]
**Package ecosystem**
ruby / bundler

**Package manager version**
Bundler version 2.1.4
**Language version**
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]

**Manifest location and content prior to update**
/Gemfile.lock - see https://github.com/jyn514/jyn514.github.io/blob/5e02d18696f849b0cc99287fa27843c158ce46f5/Gemfile.lock
**dependabot.yml content**
N/A
**Updated dependency**
nokogiri (1.10.4 to 1.13.1)
**What you expected to see, versus what you actually saw**
dependabot generates a PR that only changes the Gemfile. Instead, the PR modifies other files as well.
**Native package manager behavior**
N/A
**Images of the diff or a link to the PR, issue or logs**
https://github.com/jyn514/jyn514.github.io/pull/5

**Comment by jeffwidman** [comment]
Thanks for including direct link to the PR, makes it much easier to understand what you're describing.

When you force pushed `master`, then the existing PR branch still had the old commit that you blew away from `master`. You can see that the SHA you force pushed from is the one that's listed on the PR.

Given that you don't have a `.github/dependabot.yaml` file, that means this is a security update PR rather than our more typical version update PR's.

This one will require a little more digging, but wanted to jot down my notes from an initial look at the issue.

**Comment by jeffwidman** [comment]
I asked a teammate who told me that Dependabot only auto-rebases when there's a conflict with the branch. Auto-rebasing on every commit to the default branch would be pretty expensive from a compute standpoint, especially as right now `@dependabot rebase` re-runs the job, not a normal git rebase.

Since your force push didn't create any conflict with the PR, it never triggered teh rebase.

The workaround is to do a manual `@dependabot rebase` comment and that will trigger Dependabot to re-run and would clear it out.

**Comment by jyn514** [comment]
> I asked a teammate who told me that Dependabot only auto-rebases when there's a conflict with the branch.

This seems like the wrong behavior, though? My understanding is that if I'd merged the PR as-is, it would have reverted the changes from my force-push.

> Auto-rebasing on every commit to the default branch would be pretty expensive from a compute standpoint

It doesn't need to be on every push, only when there's a force-push.

**Comment by jeffwidman** [comment]
> It doesn't need to be on every push, only when there's a force-push.

True, we should add that to the list of conditions where we auto-rebase the PR.

**Comment by jeffwidman** [comment]
Related but not a duplicate:
* https://github.com/dependabot/dependabot-core/issues/4722

**Comment by jyn514** [comment]
@jeffwidman that's this issue, did you mean to link something else?

**Comment by jeffwidman** [comment]
Yes, fixed thanks @jyn514

**Comment by jyn514** [comment]
as far as I'm aware, this is still an issue. Issues do not go away just because they're old, they go away by being fixed.
